docs/api/baseplate/frameworks/queue_consumer/deprecated.rst

```diff
@@ -0,0 +1,144 @@
+``baseplate.frameworks.queue_consumer.deprecated``
+==================================================
+
+.. warning::
+    This way of creating a Baseplate queue consumer is deprecated in favor of
+    using a `QueueConsumerServer` and will be removed in the next major release.
+    Instructions for ugrading are included below.
+
+Upgrading to a QueueConsumerServer
+----------------------------------
+
+To start, you will be running your queue consumer as a "server" now, so it will
+use `baseplate-serve` rather than `baseplate-script` as the entrypoint.
+
+.. code-block:: diff
+
+    - baseplate-script run.ini consumer:run
+    + baseplate-serve run.ini --bind 0.0.0.0:8080
+
+This also means that you will need to update your config file with similar
+sections to what you would have for an HTTP or Thrift service.
+
+.. code-block:: diff
+
+    [DEFAULT]
+    rabbitmq.hostname = amqp://rabbit.local:5672
+    rabbitmq.exchange_name = my-exchange
+    rabbitmq.exchange_type = direct
+
+    + [app:main]
```

Comment:
maybe a name other than `main`?

Reply:
I wasn't sure about that?  I didn't want people to think that they _had_ to set an app/server name when doing this and I thought that if I put that in the example, it might give that impression.

Reply:
fair enough. the whole thing's a bit confusing anyway :(

Reply:
Ya, I can see it either way :/